### PR TITLE
[NO-TICKET] purge storage vars

### PR DESCRIPTION
**Pull request by Lkruitwagen**
### Description
- storage only needs one decision variable: `NewStorageCapacity`. Everything else can and should be linexpr
- simplified linear expressions and constraints by stacking variables and then cumulative sum

### Checklist
- [x] Dependencies install correctly in a clean environment and code executes;
- [x] Test coverage extended; created tests fail without the change (if possible);
- [x] All tests passing;
- [x] Commits follow a type convention (e.g. https://gist.github.com/brianclements/841ea7bffdb01346392c#type);
- [ ] Extended the README, documentation and/or docstrings, if necessary;

### CODE REVIEW

File: tz/osemosys/model/linear_expressions/storage.py [hunk 1]
```diff
@@ -5,96 +5,73 @@
...
+    )
+
+    ds["Conversionlh"] * ds["Conversionld"] * ds["Conversionls"]
 
     RateOfStorageCharge = (
-        (ConversionFactor * ds["TechnologyToStorage"] * m["RateOfActivity"]).where(
-            (ds["TechnologyToStorage"].notnull())
-            & (ds["TechnologyToStorage"] != 0)
-            & (ConversionFactor).notnull()
+        (ds["TechnologyToStorage"] * m["RateOfActivity"]).where(
```
Comment (Lkruitwagen) [inline tz/osemosys/model/linear_expressions/storage.py:16]: @edwardxtg I think we need some kind of scaling factor here. Maybe just needs `YearSplit`?

Comment (edwardxtg) [reply tz/osemosys/model/linear_expressions/storage.py:16]: What makes you think that? 
If we just had the conversion factor before, this was just a boolean.

File: tests/test_solve/test_solve.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def test_model_construction_from_yaml():
     model._m.solve()
 
     assert model._m.termination_condition == "optimal"
-    assert np.round(model._m.objective.value) == 10744786.0
+    assert np.round(model._m.objective.value) == 28599.0
```
[the remark this answers is not in the document]
Comment (Lkruitwagen) [reply tests/test_solve/test_solve.py:20]: Good spot @edwardxtg ! Have a look at the changes to `utopia/main.yaml`. You'll see there was a typo such that no `RL` (residential lighting) was being produced! The model was falling back to its 'unmet demand' technology which has a massive cost. If you look at the orders of magnitude in Utopia I think this objective is much closer to what I would expect. Thoughts?

File: tz/osemosys/model/linear_expressions/storage.py (hunk 1 above)
Comment (Lkruitwagen) [reply tz/osemosys/model/linear_expressions/storage.py:16]: hmm right this conversion factor was probably just casting the charge into the correct `DAILYTIME, DAYTYPE, and SEASON`. Somewhere, though, we need the actual amount of electricity generated right? Like, the `RateOfCharge` is a flux (GW etc), and somehow we need to change that into a quantity (GWh or the like). Are we doing that anywhere?

Comment (Lkruitwagen) [reply tz/osemosys/model/linear_expressions/storage.py:16]: I think, for now, if we just multiply by `YearSplit` we ca call this taken care of?

File: tests/test_solve/test_solve.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (edwardxtg) [reply tests/test_solve/test_solve.py:20]: Makes sense!

**Review by edwardxtg — APPROVED** [review]

**Pull request merged**